src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Reference.cs

```diff
@@ -258,7 +258,7 @@ public void LoadXml(XmlElement value)
                         Transform? transform = CryptoHelpers.CreateFromName<Transform>(algorithm);
                         if (transform == null)
                         {
-                            throw new CryptographicException(SR.Cryptography_Xml_UnknownTransform);
+                            throw new CryptographicException(SR.Format(SR.Cryptography_Xml_UnknownTransform_AlgorithmName, algorithm));
```

Comment:
I guess I didn't realize this when I wrote the code, but throwing an exception from `CreateXmlDsigXsltTransform()` doesn't really make sense:

https://github.com/dotnet/runtime/blob/2d9cb2d33f6f89f0a1a01782f2e474051bb2894a/src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CryptoHelpers.cs#L51-L79

The exception is always caught in `CreateFromName` and then `null` is returned.

I wonder if we could simply add a check in that catch block. Something like:

```diff
            try
            {
                return (CryptoConfig.CreateFromName(name) ?? CreateFromKnownName(name)) as T;
            }
            catch (Exception e)
            {
+                if (e is NotSupportedException && name == "http://www.w3.org/TR/1999/REC-xslt-19991116")
+                {
+                    // allow XSLT NotSupportedExceptions to be thrown
+                    throw;
+                }
                return null;
            }
```

Reply:
I refactored it to throw the exception from CreateFromName() in the first place.